util/src/main/java/io/kubernetes/client/Copy.java

```diff
@@ -115,48 +115,51 @@ public void copyDirectoryFromPod(String namespace, String pod, String srcPath, P
   public void copyDirectoryFromPod(
       String namespace, String pod, String container, String srcPath, Path destination)
       throws ApiException, IOException {
-    // TODO: Test that 'tar' is present in the container?
-    final Process proc =
-        this.exec(
-            namespace,
-            pod,
-            new String[] {"sh", "-c", "tar cz - " + srcPath + " | base64"},
-            container,
-            false,
-            false);
-    try (InputStream is = new Base64InputStream(new BufferedInputStream(proc.getInputStream()));
-        ArchiveInputStream archive = new TarArchiveInputStream(new GzipCompressorInputStream(is))) {
-      for (ArchiveEntry entry = archive.getNextEntry();
-          entry != null;
-          entry = archive.getNextEntry()) {
-        if (!archive.canReadEntryData(entry)) {
-          log.error("Can't read: " + entry);
-          continue;
-        }
-        File f = new File(destination.toFile(), entry.getName());
-        if (entry.isDirectory()) {
-          if (!f.isDirectory() && !f.mkdirs()) {
-            throw new IOException("create directory failed: " + f);
+    // Test that 'tar' is present in the container?
+    if (isTarPresentInContainer(namespace, pod, container)) {
```

Comment:
```suggestion
    if (!isTarPresentInContainer(namespace, pod, container)) {
        // do something
    }
```

nit: plz reverse the test for a cleaner indent?

Reply:
I was originally thinking of exception but wasn't sure of the name. I have created an exception CopyNotSupportedException in io.kubernetes.client.util.exception package. Hope that is right place for it. I have also flipped the if check. Also added the CopyNotSupportedException in respective calling methods.